esp-hal/src/uart.rs

```diff
@@ -508,6 +546,130 @@ where
         }
         count
     }
+
+    /// Configures the RX-FIFO threshold
+    ///
+    /// # Errors
+    /// `Err(Error::InvalidArgument)` if provided value exceeds maximum value
+    /// for SOC :
+    /// - `esp32` **0x7F**
+    /// - `esp32c6`, `esp32h2` **0xFF**
+    /// - `esp32c3`, `esp32c2`, `esp32s2` **0x1FF**
+    /// - `esp32s3` **0x3FF**
+    pub fn set_rx_fifo_full_threshold(&mut self, threshold: u16) -> Result<(), Error> {
```

Comment:
If this is part of the config now, then I don't think this should be public anymore, we should do everything through the config struct (we can still use this internally to actually set the config though).